src/vehicle_display.cpp

```diff
@@ -120,6 +120,114 @@ nc_color vehicle::part_color( const int p, const bool exact ) const
     }
 }
 
+size_t vehicle::vehicle_info( std::vector<std::string> &out_info, const uint32_t veh_part,
+                              const uint32_t max_width, const int hl /* = -1*/ )
+{
+    const size_t old_size = out_info.size();
+
+    //brief description
+    std::string message = string_format( _( "There is a %s there. Parts:" ), name );
+    foldstring( out_info, colorize( message, c_dark_gray ), max_width );
+    const size_t desc_offset = out_info.size() - old_size;
+
+    //boundary check for part id
+    if( veh_part >= parts.size() ) {
+        return 0;
+    }
+
+    std::vector<int> parts_id_list = parts_at_relative( parts[veh_part].mount, true );
+
+    for( size_t i = 0; i < parts_id_list.size(); i++ ) {
+
+        const size_t id = static_cast<size_t>( parts_id_list[i] );
+        const vehicle_part &vp = parts[id];
+        //color
+        nc_color sym_clr = c_light_gray;
+        nc_color part_clr = vp.is_broken() ? c_dark_gray : vp.get_base().damage_color();
+        //name
+        std::string partname = vp.name() + " ";
+        //fuel content
+        partname += ( vp.is_fuel_store() && vp.ammo_current() != "null" ) ? fuel_info( vp ) : "";
+        //cargo
+        partname += part_flag( id, "CARGO" ) ? cargo_info( id ) : "";
+        //armor
+        auto sym = armor_info( id );
+        //highlight selected
+        if( static_cast<const int>( i ) == hl ) {
+            sym_clr = hilite( c_light_gray );
+            part_clr = hilite( part_clr );
+        }
+
+        sym.first = colorize( sym.first, sym_clr );
+        sym.second = colorize( sym.second, sym_clr );
+        partname = colorize( partname, part_clr );
+        partname = trim_to_width( ( sym.first + partname + sym.second ), max_width );
+        out_info.push_back( partname );
+    }
+
+    //Covered or not?
+    size_t line_written = ( out_info.size() - old_size );
+    if( line_written > 0 ) {
+
+        const bool is_inside = vpart_position( *this, parts_id_list[0] ).is_inside();
+        std::string desc = is_inside ? _( "Interior" ) : _( "Exterior" );
+
+        const size_t target = out_info.size() - line_written;
```

Comment:
since out_info hasn't changed since you started this block:

`target = out_info.size() - ( out_info.size() - old_size ) == out_info.size() - out_info.size() + old_size == old_size`